ui/src/criteria/attribute.tsx

```diff
@@ -81,23 +141,149 @@ function AttributeEdit(props: AttributeEditProps) {
   const hintDisplayName = (hint: tanagra.EnumHintValue) =>
     hint.displayName || "Unknown Value";
 
-  const enumHintValues = underlay.entities
+  const attributeFilterHint = underlay.entities
     .find((g) => g.name === underlay.primaryEntity)
-    ?.attributes?.find((attribute) => attribute.name === props.data.attribute)
-    ?.attributeFilterHint?.enumHint?.enumHintValues;
+    ?.attributes?.find(
+      (attribute) => attribute.name === props.data.attribute
+    )?.attributeFilterHint;
+  const enumHintValues = attributeFilterHint?.enumHint?.enumHintValues;
+  const integerBoundsHint = attributeFilterHint?.integerBoundsHint;
 
-  const selectionIndex = (hint: tanagra.EnumHintValue) =>
-    props.data.selected.findIndex(
-      (row) => row.id === hint.attributeValue?.int64Val
-    );
+  console.log(underlay);
+  console.log(attributeFilterHint);
 
-  if (enumHintValues?.length === 0) {
+  if (!isValid(integerBoundsHint) && enumHintValues?.length === 0) {
     return (
       <Typography>
         No information for attribute {props.data.attribute}.
       </Typography>
     );
   }
+
+  if (
+    integerBoundsHint?.max !== null &&
```

Comment:
`isValid(integerBoundsHint?.min) && isValid(integerBoundsHint?.max)` ?

Reply:
This is a really strange place could not use isValid function there. or it will cause error as "TS2345: Argument of type 'number | null | undefined' is not assignable to parameter of type 'number | (() => number)'.   Type 'undefined' is not assignable to type 'number | (() => number)'. in L167-L170

I talked with Tim and he proposed to keep like that and take a look at it later and I will write a comment there to explain it.